src/App.tsx

```diff
@@ -1,19 +1,122 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  FormEvent,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import { TodoList } from './components/Auth/TodoList';
+import { Errormessage } from './components/Auth/Errormessage';
+import { Footer } from './components/Auth/Footer';
+import { Todo } from './types/Todo';
+import { FilterType } from './types/SortType';
+import { getTodos, createTodos, deleteTodos } from './api/todos';
+import { ErrorMessage } from './types/Error';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.All);
+  const [title, setTitle] = useState('');
+  const [selectedId, setSelectedId] = useState<number[]>([]);
+  const [isAdding, setIsAdding] = useState(false);
+  const user = useContext(AuthContext);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
   }, []);
 
+  useEffect(() => {
+    const getTodosFromServer = async (userId: number) => {
+      try {
+        const receivedTodos = await getTodos(userId);
+
+        setTodos(receivedTodos);
+      } catch (error) {
+        setErrorMessage(`${error}`);
+      }
+    };
+
+    if (!user) {
+      return;
+    }
+
+    getTodosFromServer(user.id);
+  }, []);
+
+  const filteredTodos = useMemo(() => todos.filter(todo => {
+    switch (filterType) {
+      case FilterType.All:
+        return todo;
+      case FilterType.Active:
+        return !todo.completed;
+      case FilterType.Completed:
+        return todo.completed;
+      default:
+        return null;
+    }
+  }), [todos]);
+
+  const newTodo = useCallback(async (event: FormEvent) => {
```

Comment:
```suggestion
  const addNewTodo = useCallback(async (event: FormEvent) => {
```
function name should describe an action - what this function do

Reply:
or `handleTodoAdding`